src/transformers/hyperparameter_search.py

```diff
@@ -119,7 +119,9 @@ def default_hp_space(self, trial):
 
 
 def default_hp_search_backend() -> str:
-    available_backends = [backend for backend in ALL_HYPERPARAMETER_SEARCH_BACKENDS.values() if backend.is_available()]
+    available_backends = [
+        backend for backend in ALL_HYPERPARAMETER_SEARCH_BACKENDS.values() if backend().is_available()
+    ]
```

Comment:
OK, it works. But I am wondering why we don't create instances like

```python
ALL_HYPERPARAMETER_SEARCH_BACKENDS = {
    HPSearchBackend(backend.name): backend() for backend_class in [OptunaBackend, RayTuneBackend, SigOptBackend, WandbBackend]
}
```

Otherwise, all the methods in the backend class could just be class method.

I personally would try to make things less confusing, but leave @sgugger to make the final call.

Reply:
Thanks @pacman100 for the fix!

Reply:
Yes should be a static method to avoid having to instantiate here.